src/main/java/jnr/posix/LinuxFileStat64.java

```diff
@@ -10,21 +10,22 @@ public Layout(jnr.ffi.Runtime runtime) {
         }
 
         public final Signed64 st_dev = new Signed64();
-        public final Signed64 st_ino = new Signed64();
+        public final Unsigned64 st_ino = new Unsigned64();
         public final Signed64 st_nlink = new Signed64();
-        public final Signed32 st_mode = new Signed32();
-        public final Signed32 st_uid = new Signed32();
-        public final Signed32 st_gid = new Signed32();
+        public final Unsigned32 st_mode = new Unsigned32();
+        public final Unsigned32 st_uid = new Unsigned32();
+        public final Unsigned32 st_gid = new Unsigned32();
+        public final Signed32 __pad0 = new Signed32();
```

Comment:
@etehtsea yeah I guess I am mostly asking because by ABI then this should have never worked until this PR.  So that makes me wonder if some Linux variants are actually missing a field that the system you tested have (which might just be a literal pad).

Unfortunately, this structure since the very beginning has not had that extra 32 bit pad.  We can probably land this as-is and write off the old code as a mistake but I will poke around and see if I can understand why there is a pad (or more importantly why we committed something without the pad).